opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -152,8 +153,6 @@ def detach(token: object) -> None:
     """
     try:
         _RUNTIME_CONTEXT.detach(token)
-    except TypeError:
-        logger.exception("Expected an instance of Token, got None")
```

Comment:
I'll keep the code as-is

Reply:
From what @rslinckx said in #4163, should the as-is codes here be included in this PR? How do you think?